src/modules/mc_pos_control/mc_pos_control_main.cpp

```diff
@@ -1766,12 +1772,6 @@ MulticopterPositionControl::calculate_velocity_setpoint(float dt)
 	vel_sp_slewrate(dt);
 	_vel_sp_prev = _vel_sp;
 
-	/* make sure velocity setpoint is constrained in all directions*/
-	if (vel_norm_xy > _vel_max_xy) {
```

Comment:
the limitation of vel max should be done the end, where no farther velocity adjustment is done